feat: deduplicate layout conversion modal.

yaldram opened the pull request:
## Description
- Deduplicate layout conversion modal.

Fixes #22312 

Media - https://www.loom.com/share/4c0352b30f83499898d4aecaf432c483


## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)


## How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.
> Delete anything that is not important

- Manual
- Jest
- Cypress

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag


### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

yaldram commented:
/build-deploy-preview env=release

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionCompleteLayout.tsx:
```diff
@@ -1,14 +1,22 @@
+  },
+  [AlertType.ERROR]: {
+    name: "error",
+    color: "var(--ads-v2-color-bg)",
```
why the error alert color is `white color` token?

yaldram replied:
The `error icon` has a gray background therefore I used the white color, the out come is as follows - 
<img width="46" alt="image" src="https://user-images.githubusercontent.com/13429535/232855364-48e83a12-8ae3-4eee-8ece-95bc4404217c.png">
Error icon on storybook - 
<img width="49" alt="image" src="https://user-images.githubusercontent.com/13429535/232855417-8c1226b2-c6f5-41e2-ad28-087195890a78.png">

rohitagarwal88 commented:
@yaldram 
Border radius is missing
<img width="729" alt="Screenshot 2023-04-18 at 3 36 44 PM" src="https://user-images.githubusercontent.com/890915/232744779-c826ade3-f646-4472-bc18-ea3db88886b6.png">



<img width="840" alt="Screenshot 2023-04-18 at 3 35 46 PM" src="https://user-images.githubusercontent.com/890915/232744534-246ab88c-9235-4f53-9916-ee7db9349a3e.png">

Border radius is missing
<img width="720" alt="Screenshot 2023-04-18 at 3 38 12 PM" src="https://user-images.githubusercontent.com/890915/232745109-d40ada04-240a-4048-958a-f46e0a508018.png">

<img width="866" alt="Screenshot 2023-04-18 at 3 37 29 PM" src="https://user-images.githubusercontent.com/890915/232744955-de9fea17-40d3-47df-978c-3bc2103593e0.png">

Border radius is missing
<img width="1496" alt="Screenshot 2023-04-18 at 3 38 39 PM" src="https://user-images.githubusercontent.com/890915/232745202-a10a8c33-be3e-4712-8f4d-dd79b8ac7219.png">


Heading isssue
https://www.loom.com/share/3e170b8f874f4910a09fb64e05ae1516

yaldram commented:
The Header title is not an issue, when the conversion completes that means when we are on the success screen we have already changed the layout, therefore the component re-renders and the title is changed.
<img width="372" alt="image" src="https://user-images.githubusercontent.com/13429535/232855985-0243a16a-4da8-470d-a049-0eca9b0dc32c.png">
We should instead show a `Conversion completed` message on the success screen.

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionButton.tsx:
```diff
@@ -22,11 +26,25 @@ import {
+import { useConversionForm } from "./hooks/useConversionForm";
+import type { AppState } from "ce/reducers";
+
+const Title = styled.h1`
```
@yaldram : We could have used a text component from ADS 2.0.

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionButton.tsx:
```diff
@@ -35,44 +53,47 @@ function ConversionButton() {
+        <ModalContent>
+          <ModalHeader>
+            <div className="flex items-center gap-3">
+              <Title>
```
@yaldram: Please do not use the title here. Any modal title should consume modal header text styles.

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionButton.tsx:
```diff
@@ -35,44 +53,47 @@ function ConversionButton() {
+            </div>
+          </ModalHeader>
+          <ModalBody>
+            <ConversionForm {...formProps} />
```
@yaldram : ConversionForm has button cancel  & convert button that should be passed in `ModalFooter`

yaldram replied:
As discussed we will have all our Modal buttons right aligned. For scenarios like the below one we decided, not to use the `Cancel` button and instead use the `Modal Header Close Button`. This will also help us achieve consistency across all modals.
![image](https://user-images.githubusercontent.com/13429535/232998021-9840451e-ed96-4f36-a8bc-f503c1fd2dd3.png)

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionForm.tsx:
```diff
@@ -1,41 +1,67 @@
+import styled from "styled-components";
 import type { InfoBlockProps } from "./InfoBlock";
 import { InfoBlock } from "./InfoBlock";
 import type { DropdownOption } from "design-system-old";
```
why do we need `DropdownOption`?

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionForm.tsx:
```diff
@@ -57,169 +83,141 @@ export type ConversionProps = {
+          <SnapshotContainer style={snapShotStyles}>
+            <Icon
+              className="mx-3"
+              color={Colors.GRAY_600}
```
Do not use Old Ads color, replace it will ADS 2.0 token

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/ConversionForm.tsx:
```diff
@@ -57,169 +83,141 @@ export type ConversionProps = {
+              name={snapShotDetails.icon}
+              size="md"
+              withWrapper
+              wrapperColor={Colors.GRAY_600_OPAQUE}
```
Do not use Old Ads color, replace it will ADS 2.0 token

rohitagarwal88 commented on app/client/src/pages/Editor/CanvasLayoutConversion/SnapShotBannerCTA.tsx:
```diff
@@ -1,70 +1,124 @@
+      <Modal onOpenChange={closeModal} open={showModal}>
+        <ModalContent>
+          <ModalHeader>
+            <ModalTitle>{modalHeader}</ModalTitle>
```
Dont override modalHeader style for title, let it be default one

rohitagarwal88 commented:
> The Header title is not an issue, when the conversion completes that means when we are on the success screen we have already changed the layout, therefore the component re-renders and the title is changed.

@yaldram Make sense, let do it. cc @marks0351 , @prsidhu , @momcilo-appsmith

yaldram commented:
/build-deploy-preview env=release

yaldram commented:
Loom Video - https://www.loom.com/share/4c0352b30f83499898d4aecaf432c483

yaldram commented:
![image](https://user-images.githubusercontent.com/13429535/233554252-a5267bd6-6c3f-487a-b777-0a6f0e31cd3c.png)

Please use sentence case wherever applicable.
- Increase the margin between the two sections of text.
- The label ‘Target canvas size’ is implemented as a separate text component. This should instead be the label prop that comes with the Input component.
- Also, the label in the primary button can be ‘Convert layout’ instead of ‘Convert’. Would be more clearer.

laveena-en commented:
Issues with Auto - Layout modal:

1. Intermediate modals are dismissible on clicking outside the modal which is breaking the UI flow
2.  `All done` modals are missing the respective titles in all flows 
![image](https://user-images.githubusercontent.com/109572422/233761839-d8fb1659-7c8d-4a36-8ee0-2f790f4c93c7.png)
3. Dropdown to be contained within the modal 
![image](https://user-images.githubusercontent.com/109572422/233761887-fdfe2f03-d780-4aec-b55c-cf1471fc321e.png)
4. `Send us a report` button to be changed to sentence case. Also, the button isn’t functional, clicking on it did nothing
5. Can we find a better placement for More details within the modal?
<img width="1338" alt="image" src="https://user-images.githubusercontent.com/109572422/233762130-86718c2e-6ab1-4400-bf18-f12cdfb33a73.png">
7. Can we change the buttons to sentence casing? `Use snapshot` -> 'Use Snapshot`, `Convert to fixed-layout` -> 'Convert to Fixed-Layout`etc? cc: @vasanth-appsmith Please advise

vasanth-appsmith commented:
**Attaching a screen for reference** 
![image](https://user-images.githubusercontent.com/92305123/233915909-7a4a1a4f-ac53-4493-9292-687885218053.png)

![image](https://user-images.githubusercontent.com/92305123/233915974-2f7d11a9-85c9-4aac-b82e-ffbaf58d4f79.png)
Reg 2: Currently, we do not have a callout with a title. Please style a callout using the error token and add it as needed.

yaldram commented:
@laveena-en 
1. Intermediate modals will not close when clicking outside or pressing of escape.
2. Added titles for all done modal screens.
3. Replaced Dropdown with a Segement Control component.
4. All buttons should be Sentence case.
5. Replaced the Error screen using new Error Screen designs.
6. All buttons should be Sentence case.

yaldram commented:
/build-deploy-preview env=release

laveena-en commented:
@yaldram Minor change, can we please remove the tooltip seen over text on the error modal? 
![image](https://user-images.githubusercontent.com/109572422/234747954-2debf86c-a029-4edd-8aa8-2773307a778c.png)
Rest of the changes LGTM.

@vasanth-appsmith Can you please re-check and give a final sign-off.
